src/components/DelegationDashboard/TrendingDelegates.tsx

```diff
@@ -82,7 +69,52 @@ const TrendingDelegates = () => {
 	useEffect(() => {
 		getData();
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [address, delegationDashboardAddress, network]);
+	}, [address, sortOption, network, selectedSources]);
```

Comment:
call this fn only initially here 
for filters and sort try to call the fn in filter change fn instead of useEffect